### PR TITLE
Theme Showcase: Show active theme as the first theme in the grid

### PR DESCRIPTION
**Pull request by taipeicoder**
#### Proposed Changes

This PR removes the current theme card from the Theme Showcase. Instead, we show the active theme as the first theme in the grid, but only when there are no queries set. This means, no search queries, no category filter selected, and no pricing filter selected. See p1669167875807189-slack-C048CUFRGFQ for more context.

| Before | After |
| --- | --- |
| Screen Shot 2022-11-23 at 6 47 43 PM | Screen Shot 2022-11-23 at 6 48 15 PM

#### Testing Instructions

* Head to the Theme Showcase `/themes/${site_slug}`.
  * If using calypso.live, the flag `themes/showcase-i4/search-and-filter` is required.
* Ensure that the current theme card is no longer present.
* Ensure that the active theme is shown first in the theme grid when no queries are set. 
* Ensure that no themes are missing because of the reordering.
* Ensure that the logged-out Theme Showcase works as before.
* Ensure that the existing Theme Showcase works as before.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/themes/theme-showcase.jsx
Comment (taipeicoder) [inline client/my-sites/themes/theme-showcase.jsx:509]: The changes in this file are unrelated to the main purpose of this PR, but fixes an issue where logged-out showcase wouldn't display category filters in the new design.

File: client/state/themes/selectors/get-themes-for-query-ignoring-page.js
Comment (arthur791004) [inline client/state/themes/selectors/get-themes-for-query-ignoring-page.js]: Could we select the active theme from the redux store instead of getting it from the theme query result? The reason is if it doesn't show on the first page, then the user won't see it at the beginning

File: client/state/themes/selectors/get-themes-for-query-ignoring-page.js
Comment (arthur791004) [inline client/state/themes/selectors/get-themes-for-query-ignoring-page.js]: [nit] I guess the checking of the `state.ui` is for the `getSelectedSiteId` function, it might be to declare the `selectedSiteId` first. Then, people won't feel confused about why we need to check the `state.ui` here at their first sight.

Comment (taipeicoder) [reply client/state/themes/selectors/get-themes-for-query-ignoring-page.js]: Makes sense 👍

File: client/state/themes/selectors/get-themes-for-query-ignoring-page.js
Comment (taipeicoder) [reply client/state/themes/selectors/get-themes-for-query-ignoring-page.js]: Even though the query fetches all themes (see filename), I've refactored the code so that we leverage data store selectors to fetch the active theme. Let me know if this approach works 🙂

**Review by arthur791004 — APPROVED** [review]
LGTM 👍

File: client/state/themes/selectors/get-themes-for-query-ignoring-page.js
Comment (fushar) [inline client/state/themes/selectors/get-themes-for-query-ignoring-page.js]:
```
themesForQueryIgnoringPage.filter =
```

What does this mean?

Comment (taipeicoder) [reply client/state/themes/selectors/get-themes-for-query-ignoring-page.js]: Good catch 🤦🏻‍♂️

**Comment by taipeicoder** [comment]
@fushar since you're also reviewing this PR, I'll wait for your approval as well before merging 🙂

**Review by fushar — APPROVED** [review]
Looks good.

Tested the logged-out flow as well!

File: client/state/themes/selectors/get-themes-for-query-ignoring-page.js
Comment (arthur791004) [reply client/state/themes/selectors/get-themes-for-query-ignoring-page.js]: Ooops... I misread it as followed 🤦‍♂️

```js
themesForQueryIgnoringPage = themesForQueryIgnoringPage.filter
```

**Pull request merged**
